core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -225,7 +223,7 @@ public static PartitionStatisticsFile computeAndWriteStatsFile(Table table, long
       }
 
       try {
-        stats = computeAndMergeStatsIncremental(table, snapshot, partitionType, statisticsFile);
+        stats = computeAndMergeStatsIncremental(table, snapshot, statisticsFile.snapshotId());
```

Comment:
Why do we remove the `partitionType` parameter? We recalculate it later again for every file. Isn't that unnecessary?

Reply:
The reason this param is removed is that `computeAndMergeStatsIncremental` is changed to use the new Scan API instead of `readPartitionStatsFile`. Since the partitionType is deliberately made internal to the Scan, this is no longer needed in this function.
What Do you mean we recalculate it later for every file? Whenever we calculate stats, we produce this partitionType once for scanning the latest existing partition stats in case it's an incremental computation, and then we produce it one more time when we sort the stats. I think this is cheap enough to worth keeping it internal to the Scan API and not giving a way for users to inject it into the scan in order to save one computation of this.

Reply:
>  we produce this partitionType once for scanning the latest existing partition stats in case it's an incremental computation, and then we produce it one more time when we sort the stats.

This is why I asked. Previous code calculated this once and reused it later